gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -683,7 +683,8 @@ public static Builder builder() {
   BucketInfo update(BucketInfo bucketInfo, BucketTargetOption... options);
 
   /**
-   * Update blob information.
+   * Update blob information. All the blob's metadata will be replaced using values from the given
+   * {@code blobInfo}. Update will fail if an incomplete {@code blobInfo} is provided.
```

Comment:
I spoke with @BrandonY as I don't understand why `update` would be any different than `create` regarding required parameters for blobs, and why `create` does not fail even when `content-type` (which is a required param for insert operation) is not provided.

I suggest creating a separate issue[s] for the above and go ahead with this PR as is and just taking action about the documentation by really making sure that people only rarely choose `update` over `patch`.  And, maybe, even taking a more extreme approach by only exposing
the `update` in `Storage`.

We could even use for method names "update" to trigger "rpc-patch" and "replace" to trigger "rpc-update".

Another, even more extreme, step would be to create a special set of options for update (similar to `BlobWriteOption`) with an option for "replace" (not visible in the rpc) that would trigger a call for "rpc-update" instead of "rpc-patch".

Lastly, see my [comment](https://github.com/GoogleCloudPlatform/gcloud-java/issues/60#issuecomment-150071558) in #60. It appears that we could use `null` to clear the metadata, so maybe we just need to fix that and add some documentation around it.